AudioBar/AudioBarViewController.swift

```diff
@@ -22,10 +22,8 @@ public final class AudioBarViewController: UIViewController, Elm.Delegate {
     @IBOutlet private var audioRouteView: UIView!
 
     @IBAction func userDidTapPlayPauseButton() {
-        switch program.view.playPauseButtonMode {
-        case .play: program.dispatch(.play)
-        case .pause: program.dispatch(.pause)
-        }
+        let mode = program.view.playPauseButtonMode
+        program.dispatch(.playPauseButton(mode))
```

Comment:
My original suggestion was about exposing a "callback" message in `View`.

This is much more complicated.